src/test/shell/integration/runfiles_test.sh

```diff
@@ -427,5 +427,32 @@ EOF
   expect_log_once "Runfiles must not contain middleman artifacts"
 }
 
+function test_removal_of_old_tempfiles() {
+  cat > BUILD << EOF
+sh_binary(
+    name = "foo",
+    srcs = ["foo.sh"],
+)
+EOF
+  touch foo.sh
+  chmod +x foo.sh
+
+  # Build once to create a runfiles directory.
+  bazel build //:foo $EXTRA_BUILD_FLAGS >&$TEST_log || fail "build failed"
+
+  # Remove the MANIFEST file that was created by the previous build.
```

Comment:
Since `build-runfiles.cc` renames `MANIFEST.tmp` into `MANIFEST` when it's done, it seems that the only way you could run into this issue is if `build-runfiles.cc` is interrupted mid-build (which can definitely happen if the Blaze server is killed). Do I understand this correctly? If so, let's add a comment here along the lines of "This simulates the case where the runfiles creation process is interrupted and leaves the temporary file behind".

Reply:
Yes, that's correct. I've extended the comment. PTAL!